src/main/java/com/github/manolo8/darkbot/backpage/GalaxyManager.java

```diff
@@ -0,0 +1,107 @@
+        gate.setPrepared(Integer.parseInt(e.attributeValue("prepared")));
+        gate.setState(e.attributeValue("state"));
+        gate.setTotal(Integer.parseInt(e.attributeValue("total")));
+        gate.setTotalWave(Integer.parseInt(e.attributeValue("totalWave")));
+        return gate;
+    }
+
+    private static EnergyCost analyzeEnergyCost(Element e) {
+        EnergyCost energyCost = new EnergyCost();
+
+        energyCost.setMode(e.attributeValue("mode"));
+        energyCost.setValue(Integer.parseInt(e.getText()));
+        return energyCost;
+    }
+
+
+    private static Multiplier analyzeMultipler(Element e) {
+        Multiplier multiplier = new Multiplier();
+
+        multiplier.setMode(e.attributeValue("mode"));
+        multiplier.setState(Integer.parseInt(e.attributeValue("state")));
+        multiplier.setValue(Integer.parseInt(e.attributeValue("value")));
+        return multiplier;
+    }
+
+    public void updateJumpgateInfo() {
+        String params = "flashinput/galaxyGates.php?userID=" + main.hero.id + "&action=init&sid=" + main.statsManager.sid;
+        updateJumpgateInfo(params, 100);
+    }
+
+    public void updateJumpgateInfo(String params, int minWait) {
+        List<Gate> gates = new ArrayList<>();
+        List<EnergyCost> energyCosts = new ArrayList<>();
+        List<Multiplier> multipliers = new ArrayList<>();
+        List<Item> items = new ArrayList<>();
+
+        Element root = getRootElement(params, minWait);
+
+        List<Element> gateEle = root.elementIterator("gates").next().elements();
+        List<Element> multiplersEle = root.elementIterator("multipliers").next().elements();
+        //List<Element> itemsEle = root.elementIterator("items").next().elements();
+        energyCosts.add(analyzeEnergyCost(root.element("energy_cost")));
+
+        for (Element e : gateEle) {
+            Gate gate = analyzeGate(e);
+            gates.add(gate);
+        }
+        for (Element e : multiplersEle) {
+            Multiplier multiplier = analyzeMultipler(e);
+            multipliers.add(multiplier);
+        }
+
+        jumpgate.setGates(gates);
+        jumpgate.setEnergyCosts(energyCosts);
+        jumpgate.setMultipliers(multipliers);
+        jumpgate.setBonusRewardsDay(Integer.parseInt(root.element("bonusRewardsDay").getText()));
+        jumpgate.setMoney(Integer.parseInt(root.element("money").getText()));
+        jumpgate.setSamples(Integer.parseInt(root.element("samples").getText()));
+        jumpgate.setGalaxyGateDay(Integer.parseInt(root.element("galaxyGateDay").getText()));
+        jumpgate.setSpinOnSale(Integer.parseInt(root.element("spinOnSale").getText()));
+        jumpgate.setSpinSalePercentage(Integer.parseInt(root.element("spinSalePercentage").getText()));
+    }
+
+    private Element getRootElement(String params, int minWait) {
+        Element element = null;
+        try {
+            SAXReader reader = new SAXReader();
```

Comment:
Are you sure you should't be using a DOM reader for this instead? You're using a document and not events